src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelist.java

```diff
@@ -92,11 +115,13 @@ public final void reset(Collection<? extends Whitelist> delegates) {
                     staticMethodSignatures.addAll(ew.staticMethodSignatures());
                     fieldSignatures.addAll(ew.fieldSignatures());
                     staticFieldSignatures.addAll(ew.staticFieldSignatures());
+                    ew.clearCache();
```

Comment:
I do not actually think you need to do this. Just set up the cache in the `EnumeratingWhitelist` initializer and never touch it again.

Reply:
See answer below -- we're avoiding redundant cache generation.